### ISSUE TITLE
Expose object identity for all debugger object views

### ISSUE THREAD

**Issue by jacob314** [flutter/devtools#2772]
Seeing the identity hash helps users tell which objects they see in multiple places are the same.

Fyi @bkonyi, I expect we can cheaply expose the identity hash for all InstanceRef objects that are not primitive types as we already are for the snapshots.

**Comment by grouma** [comment]
Any update here?

**Comment by jacob314** [comment]
Actually the real sticking point is probably the web. We would want to add identity hashes to each Dart object on the web side without making an extra roundtrip to JavaScript. If that isn't practical, we should expose the identity hash on each Instance object rather than each InstanceRef object. That should still be sufficient for the DevTools debugger.

**Comment by grouma** [comment]
I don't think this will be an issue on the web side. At the end of the day everything is backed by a [RemoteObject](https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#type-RemoteObject) which has a unique ID associated with it. We could us this identifier on both the InstanceRef and actual Instance objects.

**Comment by kenzieschmoll** [comment]
Closing this as not planned, since the DevTools debugger is in maintenance mode. We recommend that users use a more full-featured debugger from their IDE.
